compiler/src/iree/compiler/Dialect/HAL/Target/ROCM/ROCMTarget.cpp

```diff
@@ -229,9 +286,15 @@ class ROCMTargetBackend final : public TargetBackend {
       }
       linkROCDLIfNecessary(llvmModule.get(), clROCMTargetChip, rocmBitcodeDir);
     }
+    // Add Optimize module
+    optimizeModule(*llvmModule, *targetMachine);
 
     // Serialize hsaco kernel into the binary that we will embed in the
     // final FlatBuffer.
+    std::unique_ptr<llvm::Module> moduleCopy;
+    if (!options.dumpIntermediatesPath.empty()) {
+      moduleCopy = llvm::CloneModule(*llvmModule);
```

Comment:
could we test if the copy was successful like here?
https://github.com/ROCmSoftwarePlatform/triton/blob/triton-mlir/lib/Target/HSACO/HSACOTranslation.cpp#L264-L267

Reply:
Just updated.